src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -240,7 +230,7 @@ class BookPaymentPolicy
     }
 
     Quality
-    qualityUpperBound(ReadView const& v,
+    qualUpperBound(ReadView const& v,
```

Comment:
I don't like having a function called `qualityUpperBound` in the base and `qualUpperBound` in the derived. I'd bring in the base's implementation in with a `using BookStep<TIn, TOut, BookPaymentStep<TIn, TOut>>::qualityUpperBound;` (so we don't hide it) and just overload on `qualityUpperBound`. If you object to that, I'd suggest `qualityUpperBoundHelper` or somesuch.